diff --git a/apps/ymir-sdl3/src/app/ui/views/settings/general_settings_view.cpp b/apps/ymir-sdl3/src/app/ui/views/settings/general_settings_view.cpp
index fc341a36..adfd0264 100644
--- a/apps/ymir-sdl3/src/app/ui/views/settings/general_settings_view.cpp
+++ b/apps/ymir-sdl3/src/app/ui/views/settings/general_settings_view.cpp
@@ -156,24 +156,24 @@ void GeneralSettingsView::Display() {
                 } else {
                     ImGui::Text("%s", ver->version.to_string().c_str());
                 }
-            } else {
-                ImGui::TextUnformatted("Not checked");
-            }
-
-            ImGui::TableNextColumn();
-            if (!ver->releaseNotesURL.empty()) {
-                if (ImGui::SmallButton("Release notes")) {
-                    SDL_OpenURL(ver->releaseNotesURL.c_str());
-                }
-            }
-            if (!ver->downloadURL.empty()) {
+                ImGui::TableNextColumn();
                 if (!ver->releaseNotesURL.empty()) {
-                    ImGui::SameLine();
+                    if (ImGui::SmallButton("Release notes")) {
+                        SDL_OpenURL(ver->releaseNotesURL.c_str());
+                    }
                 }
-                if (ImGui::SmallButton("Download")) {
-                    SDL_OpenURL(ver->downloadURL.c_str());
+                if (!ver->downloadURL.empty()) {
+                    if (!ver->releaseNotesURL.empty()) {
+                        ImGui::SameLine();
+                    }
+                    if (ImGui::SmallButton("Download")) {
+                        SDL_OpenURL(ver->downloadURL.c_str());
+                    }
                 }
+            } else {
+                ImGui::TextUnformatted("Not checked");
             }
+
             ImGui::PopID();
         };
         version("Stable", m_context.updates.latestStable);
diff --git a/apps/ymir-sdl3/src/app/ui/windows/update_onboarding_window.cpp b/apps/ymir-sdl3/src/app/ui/windows/update_onboarding_window.cpp
index c48194a6..468a48da 100644
--- a/apps/ymir-sdl3/src/app/ui/windows/update_onboarding_window.cpp
+++ b/apps/ymir-sdl3/src/app/ui/windows/update_onboarding_window.cpp
@@ -41,7 +41,8 @@ void UpdateOnboardingWindow::DrawContents() {
         m_context.displayScale);
 
     ImGui::NewLine();
-    ImGui::TextUnformatted("Choose Accept to apply these settings or Decide later to close this window now.");
+    ImGui::TextUnformatted("Choose Accept to apply these settings or Decide later to close this window now.\n"
+                           "If you choose to decide later, this popup will appear again on next startup.");
 
     ImGui::Separator();
     if (ImGui::Button("Accept")) {
